### ISSUE TITLE
webpack 5 support

### ISSUE THREAD

**Issue by hmnd** [vuetifyjs/vuetify-loader#136]
In webpack/webpack#9138, the RuleSet class being used in [lib/plugin.js](https://github.com/vuetifyjs/vuetify-loader/blob/master/lib/plugin.js) has been refactored and no longer exists. This makes webpack 5 throw a MODULE_NOT_FOUND error when trying to build.

**Comment by KaelWD** [comment]
Related: https://github.com/vuejs/vue-loader/pull/1613, https://github.com/JetBrains/svg-sprite-loader/pull/403

**Comment by draeron** [comment]
any knowns a workaround for this ?

**Comment by JorisAerts** [comment]
Based on the changes on vue-loader, I also modified the plugin.js and created a plugin-webpack4 and plugin-webpack5 and it seems to work at first sight. 

So, is there already someone in the process of fixing this? Is there an official fix pendig?

**Comment by KaelWD** [comment]
If you've made the changes already feel free to open a pull request.

**Comment by bloodlit** [comment]
Add webpack 5 support #160

**Comment by chamboug** [comment]
@bloodlit How is your PR different from mine (https://github.com/vuetifyjs/vuetify-loader/pull/158) created 2 days ago?

**Comment by grbd** [comment]
Any news on this?, this would be useful since webpack 5 has inbuilt support for yarn2's pnp
and we have at least 2 pull requests in now for the same thing

**Comment by MaxMls** [comment]
The last commit was 3 months ago, perhaps author is currently busy with the transition to version 3 and cannot accept pull requests

**Comment by grbd** [comment]
I think I'll need to go back to webpack4 anyways since there's some issues when using web-dev-server with webpack5 in combination with yarn v2 pnp

https://github.com/webpack/webpack-dev-server/issues/2820

**Comment by alekbless** [comment]
I want to use this with the new Laravel Mix 6 and webpack 5 support is needed.

**Comment by bloodlit** [comment]
@alekbless take the lib folder from the branch  add-webpack5-support

**Comment by martinoppitz** [comment]
the same story here:
- https://github.com/vuetifyjs/vuetify-loader/issues/136 and here
- https://github.com/aurelia/aurelia/issues/982 and here
- https://github.com/sveltejs/svelte-loader/pull/136 and so on

Many thanks to Angular for influencing Webpack and the unconsolidated upgrade to v5.

**Comment by glen-84** [comment]
> 
> 
> Related: [vuejs/vue-loader#1613](https://github.com/vuejs/vue-loader/pull/1613), [JetBrains/svg-sprite-loader#403](https://github.com/JetBrains/svg-sprite-loader/pull/403)

Both of these PRs have been merged and released.

**Comment by bkhatkov** [comment]
Hi guys. Is there any change to get this done? It seems much easier to switch to any other lib than to wait here. Vue added support of WP5 long time ago. Libs should follow as soon as possible. We invested quite a lot to develop our apps using vuetify and now we're just stuck...

**Comment by drrrMikado** [comment]
is there any progress?

**Comment by AGDholo** [comment]
If you want to use the new es feature, you will have to update vue/cli to version 5, because webpack 4 has issues that prevent vue/cli4 from using these features, e.g. ?

But vuetify only supports webpack4, so it's hard to keep up with modern development.

**Comment by adiramardiani** [comment]
@AGDholo 

> But vuetify only supports webpack4, so it's hard to keep up with modern development.

In documentation vuetify not use `only supports` but `version >=4 of Webpack`
![image](https://user-images.githubusercontent.com/8703419/103600450-02e21300-4f3a-11eb-8241-913ddb5165b2.png)

**Comment by AGDholo** [comment]
> @AGDholo
> 
> > But vuetify only supports webpack4, so it's hard to keep up with modern development.
> 
> In documentation vuetify not use `only supports` but `version >=4 of Webpack`
> ![image](https://user-images.githubusercontent.com/8703419/103600450-02e21300-4f3a-11eb-8241-913ddb5165b2.png)

@adiramardiani 
I installed vue cli 5 (webpack v5) and added vuetify, which prompts an error and I had to use the compatibility solution (installing webpack v4 again) to get it to work: https://github.com/vuejs/vue-cli/releases/tag/ v5.0.0-alpha.0

**Comment by AGDholo** [comment]
> In documentation vuetify not use `only supports` but `version >=4 of Webpack`

@adiramardiani 

I installed vue cli 5 (webpack v5) and added vuetify, which prompts an error and I had to use the compatibility solution (installing webpack v4 again) to get it to work: https://github.com/vuejs/vue-cli/releases/tag/ v5.0.0-alpha.0

**Comment by vstyler96** [comment]
I can't still solve my problem, I am still getting this: 
![image](https://user-images.githubusercontent.com/46312870/104256249-7acfa200-5440-11eb-9856-19c346f19326.png)

**Comment by awill1988** [comment]
I have a PR #166 which is working for me, it mimics to the `plugin-webpack4/5.js` approach that exists in vue-loader. I think this gets us over the finish line right?

**Comment by vstyler96** [comment]
@awill1988 did you check this with laravel mix?

**Comment by awill1988** [comment]
I did not

**Comment by vstyler96** [comment]
@awill1988 are the custom variables working from webpack config? :o

**Comment by awill1988** [comment]
I'm not sure I'm about to publish an example with Vue2 Vuetify with module federation so that is my focus.. this seemed like the easiest lift

**Comment by KaelWD** [comment]
[v1.7.0](https://github.com/vuetifyjs/vuetify-loader/releases/tag/v1.7.0) has been published as `vuetify-loader@next`. If you find any problems with webpack 5 please report them in a new issue.

**Comment by tux2nicolae** [comment]
Just updated to vuetify-loader@next and getting this

![image](https://user-images.githubusercontent.com/39991250/104954473-f6959580-59d0-11eb-9bd0-586e87d32ce9.png)

**Comment by mathe42** [comment]
> If you find any problems with webpack 5 please report them in a new issue.

Create a new issue for that! With a reproduction repro!

**Comment by stefanKuijers** [comment]
@tux2nicolae Could you link back to this issue if you create a new issue? Just so I can follow the progress as well. :)

**Comment by tux2nicolae** [comment]
@stefanKuijers Actually I didn't created a new issues as the problem may be fixed in `1.7.1` it works for me.

**Comment by bhanuunrivalled** [comment]
HI , Iam using `"webpack": "^3.8.1"  `

```
  "vue": "^2.5.9",
    "vue-chart-js": "^1.4.1",
    "vue-chartkick": "^0.2.0",
    "vue-i18n": "^7.3.2",
    "vuetify": "^0.16.4"
```

is it possible for me to update to webpack 5?  or webpack 4? which one is better?

**Comment by kpturner** [comment]
> @stefanKuijers Actually I didn't created a new issues as the problem may be fixed in `1.7.1` it works for me.

Why do we think this is fixed in `1.7.1`?  That version still references `webpack/lib/RuleSet` in plugin.js and that does not exist in webpack 5
